lib/Transforms/MemoryBanking.cpp

```diff
@@ -179,6 +182,8 @@ struct BankAffineLoadPattern
     auto defaultValue = rewriter.create<arith::ConstantOp>(loc, zeroAttr);
     rewriter.create<scf::YieldOp>(loc, defaultValue.getResult());
 
+    if (isa<BlockArgument>(loadOp.getMemref()))
```

Comment:
> We add loadOp.memref() to oldMemRefVals only when the value is a BlockArgument because only the BlockArgument will remain in the cleanUpOldMemRefs function; and the values that are not BlockArgument will all gets replaced and naturally no longer exists, thus not requiring any additional clean-ups.

It probably would be good to add this as a (bit more concise) comment here. It isn't clear otherwise to someone reading the code.